pandas/_libs/window/aggregations.pyx

```diff
@@ -1068,60 +1068,45 @@ cdef _roll_min_max_variable(ndarray[numeric] values,
         # The original impl didn't deal with variable window sizes
         # So the code was optimized for that
 
-        for i in range(starti[0], endi[0]):
-            ai = init_mm(values[i], &nobs, is_max)
-
-            # Discard previous entries if we find new min or max
-            if is_max:
-                while not Q.empty() and ((ai >= values[Q.back()]) or
-                                         values[Q.back()] != values[Q.back()]):
-                    Q.pop_back()
-            else:
-                while not Q.empty() and ((ai <= values[Q.back()]) or
-                                         values[Q.back()] != values[Q.back()]):
-                    Q.pop_back()
-            Q.push_back(i)
-            W.push_back(i)
-
-        # if right is open then the first window is empty
-        close_offset = 0 if endi[0] > starti[0] else 1
         # first window's size
         curr_win_size = endi[0] - starti[0]
+        # GH 32865
+        # Anchor output index to values index to provide custom
+        # BaseIndexer support
+        for i in range(N):
```

Comment:
Mind running the ASVs for rolling max/min to see if the existing functions don't suffer a performance hit?

Reply:
@mroeschke
Sure! Sorry for missing that we have a benchmark for this stuff in the suite.

Ran the benchmarks in `rolling.Methods`. No significant changes.

Detailed output is below.

<details>
```
[100.00%] ··· rolling.Methods.time_rolling                                                                            ok
[100.00%] ··· ============= ======== ======= ======== =============
               constructor   window   dtype   method
              ------------- -------- ------- -------- -------------
                DataFrame      10      int    median     74.0±1ms
                DataFrame      10      int     mean     13.7±0.1ms
                DataFrame      10      int     max      9.12±0.1ms
                DataFrame      10      int     min      9.09±0.2ms
                DataFrame      10      int     std      16.2±0.2ms
                DataFrame      10      int    count      15.3±1ms
                DataFrame      10      int     skew     15.1±0.1ms
                DataFrame      10      int     kurt     15.5±0.3ms
                DataFrame      10      int     sum      13.3±0.6ms
                DataFrame      10     float   median     71.4±2ms
                DataFrame      10     float    mean     7.94±0.6ms
                DataFrame      10     float    max      8.96±0.3ms
                DataFrame      10     float    min      8.88±0.1ms
                DataFrame      10     float    std       12.0±3ms
                DataFrame      10     float   count      16.1±2ms
                DataFrame      10     float    skew     10.1±0.4ms
                DataFrame      10     float    kurt     10.4±0.7ms
                DataFrame      10     float    sum      8.19±0.6ms
                DataFrame     1000     int    median    147±0.9ms
                DataFrame     1000     int     mean      11.3±1ms
                DataFrame     1000     int     max      9.02±0.1ms
                DataFrame     1000     int     min     9.06±0.08ms
                DataFrame     1000     int     std      12.2±0.9ms
                DataFrame     1000     int    count      19.9±5ms
                DataFrame     1000     int     skew      12.7±1ms
                DataFrame     1000     int     kurt      9.19±3ms
                DataFrame     1000     int     sum       11.2±1ms
                DataFrame     1000    float   median     143±2ms
                DataFrame     1000    float    mean     3.99±0.6ms
                DataFrame     1000    float    max      8.90±0.1ms
                DataFrame     1000    float    min      9.01±0.1ms
                DataFrame     1000    float    std       11.9±3ms
                DataFrame     1000    float   count      19.8±5ms
                DataFrame     1000    float    skew     6.19±0.9ms
                DataFrame     1000    float    kurt     5.73±0.8ms
                DataFrame     1000    float    sum      3.61±0.7ms
                  Series       10      int    median     77.0±1ms
                  Series       10      int     mean     16.3±0.5ms
                  Series       10      int     max      10.3±0.2ms
                  Series       10      int     min      10.4±0.2ms
                  Series       10      int     std      19.2±0.3ms
                  Series       10      int    count      28.8±4ms
                  Series       10      int     skew     18.4±0.4ms
                  Series       10      int     kurt     18.0±0.4ms
                  Series       10      int     sum      16.7±0.2ms
                  Series       10     float   median    72.8±0.7ms
                  Series       10     float    mean     8.39±0.6ms
                  Series       10     float    max      10.4±0.1ms
                  Series       10     float    min      10.3±0.1ms
                  Series       10     float    std      15.3±0.1ms
                  Series       10     float   count      28.6±4ms
                  Series       10     float    skew      11.3±1ms
                  Series       10     float    kurt      11.9±1ms
                  Series       10     float    sum      8.23±0.6ms
                  Series      1000     int    median     151±2ms
                  Series      1000     int     mean     15.0±0.3ms
                  Series      1000     int     max      10.2±0.1ms
                  Series      1000     int     min      10.3±0.2ms
                  Series      1000     int     std      15.7±0.3ms
                  Series      1000     int    count      28.5±4ms
                  Series      1000     int     skew     16.2±0.3ms
                  Series      1000     int     kurt     16.5±0.2ms
                  Series      1000     int     sum      14.3±0.4ms
                  Series      1000    float   median     153±5ms
                  Series      1000    float    mean     14.8±0.3ms
                  Series      1000    float    max      10.2±0.1ms
                  Series      1000    float    min      10.4±0.2ms
                  Series      1000    float    std      17.3±0.3ms
                  Series      1000    float   count      28.6±4ms
                  Series      1000    float    skew     16.3±0.3ms
                  Series      1000    float    kurt     16.1±0.3ms
                  Series      1000    float    sum      14.4±0.2ms
              ============= ======== ======= ======== =============


BENCHMARKS NOT SIGNIFICANTLY CHANGED.
```
</details>

Reply:
Great thanks!